### PR TITLE
RELATED: RAIL-3492 - Add handler for change insight widget filter settings

### PR DESCRIPTION
**Pull request by lupko**
- Several prerequisites (separate commits)
- The command/event API for the filter settings overhauled big time in order to provide decent DX
- Impl & tests

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4752/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6385/ : SUCCESS in 29m 37s

File: libs/sdk-ui-dashboard/tsconfig.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
         "resolveJsonModule": true,
         "esModuleInterop": true,
         "importHelpers": true,
+        "downlevelIteration": true,
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/tsconfig.json:14]: Why is this needed?

**Review by no23reason — APPROVED** [review]
LGTM

File: libs/sdk-ui-dashboard/tsconfig.json (hunk 1 above)
Comment (lupko) [reply libs/sdk-ui-dashboard/tsconfig.json:14]: the filterValidation uses for-of loop to go through ObjRefMap.values() (which is iterator same as is in general Map contract)

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1610

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1343/ : FAILURE in 3m 35s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1619/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/400/ : SUCCESS in 15m 22s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6388/ : SUCCESS in 16m 27s
